src/phylum/ci/depfile.py

```diff
@@ -192,9 +193,9 @@ def parse_depfile(
     LOG.debug("Using parse command: %s", shlex.join(cmd))
     LOG.debug("Running command from: %s", start)
     result = subprocess.run(cmd, cwd=start, check=True, capture_output=True, text=True).stdout.strip()  # noqa: S603
-    parsed_pkgs = json.loads(result)
-    depfile_packages = [PackageDescriptor(**pkg) for pkg in parsed_pkgs]
-    return depfile_packages
+    parsed_pkgs: list[dict[str, str]] = json.loads(result)
+    depfile_pkgs_with_origin = [Package(origin=depfile_relpath, **pkg) for pkg in parsed_pkgs]
```

Comment:
BLUF: The `lockfile` field was previously set and was used/needed to show the path to the dependency file in the UI:

<img width="1320" alt="image" src="https://github.com/phylum-dev/phylum-ci/assets/18729796/fee14a22-d902-45bf-acbb-8ea187b9831f">

---

The changes from `PackageDescriptor` to `Package` account for the various CLI changes so the class can be used for a number of situations:

---

### Keeping track of packages returned by the `phylum parse` subcommand

This is the `PackageDescriptor` data structure in the CLI. It used to provide `lockfile` results to show the relative path to the dependency file but was updated from `PackageDescriptorAndLockfile` to `PackageDescriptor`, which removed the `lockfile` result.

<details><summary>Details</summary>
<p>

```
❯ cat requirements.txt
pyyaml

❯ phylum --version
phylum v5.9.0

❯ phylum parse --lockfile-type pip ./requirements.txt
Generating lockfile for manifest "requirements.txt" using Pip…
[
  {
    "name": "PyYAML",
    "version": "6.0.1",
    "type": "pypi",
    "lockfile": "requirements.txt"
  }
]

❯ phylum update --prerelease

    phylum-cli installer

    OK Copied completions to /Users/maxrake/.local/share/phylum/completions
Installing extension bundle...
Another version of the 'bundle' extension is already installed. Overwrite? yes
✅ Extension bundle installed successfully
Installing extension cargo...
Another version of the 'cargo' extension is already installed. Overwrite? yes
✅ Extension cargo installed successfully
Installing extension npm...
Another version of the 'npm' extension is already installed. Overwrite? yes
✅ Extension npm installed successfully
Installing extension pip...
✅ Extension pip already installed, nothing to do
Installing extension poetry...
✅ Extension poetry already installed, nothing to do
Installing extension yarn...
Another version of the 'yarn' extension is already installed. Overwrite? yes
✅ Extension yarn installed successfully
    OK Installed default extensions
    OK Completions are enabled for bash.
    OK Completions are enabled for zsh.
    OK Successfully installed phylum.

    Source your /Users/maxrake/.zshrc file, add /Users/maxrake/.local/bin to your $PATH variable, or
    log in to a new terminal in order to make `phylum` available.

✅ Successfully updated to v6.0.0-rc1!

❯ phylum --version
phylum v6.0.0-rc1

❯ phylum parse --type pip ./requirements.txt
Generating lockfile for manifest "requirements.txt" using Pip…
[
  {
    "name": "PyYAML",
    "version": "6.0.1",
    "type": "pypi"
  }
]
```

</p>
</details> 

---

### Representing "current" packages when submitting to Phylum CLI Extension API `analyze` call

This makes use of the new `PackageWithOrigin` CLI data structure. The `origin` attribute has a `lockfile` alias and is optional, but if not present, the path will not be shown in the UI.

---

### Representing the "base" packages used to filter results when getting a job's status

This is for the Phylum CLI extension API call `PhylumApi::getJobStatus`, which expects a list of `Package` types as defined in the extension API. This type does not include the `origin` but it also doesn't care when it gets included...which is why this updated `Package` dataclass in `phylum-ci` is used instead of creating a separate one with the `origin` field removed.

Reply:
> ### Keeping track of packages returned by the phylum parse subcommand
> This is the `PackageDescriptor` data structure in the CLI. It used to provide `lockfile` results to show the relative path to the dependency file but was updated from `PackageDescriptorAndLockfile` to `PackageDescriptor`, which removed the `lockfile` result.

This seems like a significant regression... Was this intentional? Cc: @cd-work

Reply:
The location was just moved from each package to the top of the object. No information should be lost afaik.

Reply:
It definitely appears to be missing now. I've opened phylum-dev/cli#1316 to track.

Reply:
This PR has been updated to account for the changes made to the CLI in phylum-dev/cli#1317 and released in v6.0.0-rc3